types/react-leaflet/index.d.ts

```diff
@@ -320,7 +320,7 @@ export interface LayersControlProps extends LayersControlEvents, Leaflet.Control
     children?: Children;
     overlays?: Leaflet.Control.LayersObject;
 }
-export class LayersControl<P extends AttributionControlProps, E extends Leaflet.Control.Layers> extends MapControl<P, E> { }
+export class LayersControl<P extends AttributionControlProps = LayersControlProps, E extends Leaflet.Control = Leaflet.Control.Layers> extends MapControl<P, E> { }
```

Comment:
`P` should extend `LayersControlProps`, and `E` should extend `Leaflet.Control.Layers`.